src/pdsparsemat.jl

```diff
@@ -2,24 +2,34 @@
 Sparse positive definite matrix together with a Cholesky factorization object.
 """
 struct PDSparseMat{T<:Real,S<:AbstractSparseMatrix} <: AbstractPDMat{T}
-    dim::Int
     mat::S
     chol::CholTypeSparse
 
-    PDSparseMat{T,S}(d::Int,m::AbstractSparseMatrix{T},c::CholTypeSparse) where {T,S} =
-        new{T,S}(d,m,c) #add {T} to CholTypeSparse argument once #14076 is implemented
+    function PDSparseMat{T,S}(d::Int,m::AbstractSparseMatrix{T},c::CholTypeSparse) where {T,S}
+        LinearAlgebra.checksquare(m) == d ||
+            throw(DimensionMismatch("Dimensions of mat and chol are inconsistent."))
```

Comment:
Can we make the error message more accurate and actually check the size of `chol` as well?